game/scripts/vscripts/components/progression/hero_progression.lua

```diff
@@ -5,96 +5,114 @@ if HeroProgression == nil then
     ChatCommand:LinkCommand("-levelup", "OnLevelUpChatCmd", HeroProgression)
 end
 
+GameEvents:OnPlayerLevelUp(function(keys)
```

Comment:
For some reason the event listeners seem to reset when `script_reload` is called. So having these in init made this component pretty much stop working whenever `script_reload` was done.